src/core/unif.ml

```diff
@@ -33,6 +33,38 @@ let pp_problem oc p =
 let no_problems : problems =
   {to_solve  = []; unsolved = []; recompute = false}
 
+(** Definition of symbols used to encode unification rules. *)
+module Hint = struct
+
```

Comment:
Here you need to create a signature with `Sign.create`, and immediately register it into the `Sign.loaded`. So you need to pick a unique module path for your ghost signature. Maybe we should reserve module paths starting with a an empty member? So maybe use `[""; "hint"]`?

Once you have done that you need to register your symbols using `Sign.add_symbol` (or to manually insert them if the function is not suitable for some reason). Doing all this will ensure correct linking by `Sign.link`.

I think you might need to make sure that the "ghost signature" is always open in the case where its symbols can be entered by the user. But you may also go around that in some other way. I am not sure. Is the user intended to have access to these symbols actually?

Reply:
No, I don't think the user should have access to these symbols. Or I don't see the need.

Reply:
Then I am confused about this `Syntax.Unif_hints` module. What is it for?

Reply:
Ah it must be completely useless with my last commit.